tests/src/core/testqgstemporalnavigationobject.cpp

```diff
@@ -0,0 +1,159 @@
+ ***************************************************************************/
+
+#include "qgstest.h"
+#include <QObject>
+
+//qgis includes...
+#include <qgstemporalnavigationobject.h>
+
+/**
+ * \ingroup UnitTests
+ * This is a unit test for the QgsTemporalNavigationObject class.
+ */
+class TestQgsTemporalNavigationObject : public QObject
+{
+    Q_OBJECT
+
+  public:
+    TestQgsTemporalNavigationObject() = default;
+
+  private slots:
+    void initTestCase();// will be called before the first testfunction is executed.
+    void cleanupTestCase();// will be called after the last testfunction was executed.
+    void init(); // will be called before each testfunction is executed.
+    void cleanup(); // will be called after every testfunction.
+
+    void animationState();
+    void temporalExtents();
+    void frameSettings();
+
+  private:
+    QgsTemporalNavigationObject *navigationObject = nullptr;
+};
+
+void TestQgsTemporalNavigationObject::initTestCase()
+{
+  //
+  // Runs once before any tests are run
+  //
+  // init QGIS's paths - true means that all path will be inited from prefix
+  QgsApplication::init();
+  QgsApplication::initQgis();
+  QgsApplication::showSettings();
+
+}
+
+void TestQgsTemporalNavigationObject::init()
+{
+  //create some objects that will be used in all tests...
+  //create a temporal object that will be used in all tests...
+
+  navigationObject = new QgsTemporalNavigationObject();
+}
+
+void TestQgsTemporalNavigationObject::cleanup()
+{
+}
+
+void TestQgsTemporalNavigationObject::cleanupTestCase()
+{
+  QgsApplication::exitQgis();
+}
+
+void TestQgsTemporalNavigationObject::animationState()
+{
+  qRegisterMetaType<QgsTemporalNavigationObject::AnimationState>( "AnimationState" );
+  QSignalSpy stateSignal( navigationObject, &QgsTemporalNavigationObject::stateChanged );
+
+  QCOMPARE( navigationObject->animationState(), QgsTemporalNavigationObject::Idle );
+
+  navigationObject->setAnimationState( QgsTemporalNavigationObject::Forward );
+  QCOMPARE( navigationObject->animationState(), QgsTemporalNavigationObject::Forward );
+  QCOMPARE( stateSignal.count(), 1 );
+
+  navigationObject->playBackward();
+  QCOMPARE( navigationObject->animationState(), QgsTemporalNavigationObject::Reverse );
+  QCOMPARE( stateSignal.count(), 2 );
+
+  navigationObject->playForward();
+  QCOMPARE( navigationObject->animationState(), QgsTemporalNavigationObject::Forward );
+  QCOMPARE( stateSignal.count(), 3 );
+
+  navigationObject->pause();
+  QCOMPARE( navigationObject->animationState(), QgsTemporalNavigationObject::Idle );
+  QCOMPARE( stateSignal.count(), 4 );
+
+  navigationObject->next();
+  QCOMPARE( navigationObject->currentFrameNumber(), 1 );
+
+  navigationObject->previous();
+  QCOMPARE( navigationObject->currentFrameNumber(), 0 );
+
+  navigationObject->skipToEnd();
+  QCOMPARE( navigationObject->currentFrameNumber(), navigationObject->totalFrameCount() - 1 );
```

Comment:
I don't think this is testing anything at the moment, because the default range of a navigation object is 0 length. I think you should manually set a range and time step at the start of this test and check a static value here.

Reply:
Sure

Reply:
That's better, thanks!

Reply:
Cool!